File: ingyu1008/제일작은수제거하기.cpp
#include <vector>

std::vector<int> solution(std::vector<int> arr)
{
    std::vector<int> answer;

    int min = 2147483647;
    for (auto it = arr.begin(); it != arr.end(); it++)
    {
        min = std::min(min, *it);
    }

    for (auto it = arr.begin(); it != arr.end(); it++)
    {
        answer.push_back(*it);
        if (*it == min)
        {
            min = 2147483647;
            answer.pop_back();
        }
    }
    if (answer.empty())
        answer.push_back(-1);

    return answer;
}

File: bsw112/2021-01-27/제일작은수제거하기.java
import java.util.Arrays;

public class 제일작은수제거하기 {

    public int[] solution(int[] arr) {
        if(arr.length <= 1)
            return new int[]{-1};
        int min = Arrays.stream(arr).min().getAsInt();
        return Arrays.stream(arr).filter(it -> it != min).toArray();
    }
}
